Source/ChangeTracker/CBLSocketChangeTracker.m

```diff
@@ -96,13 +96,13 @@ - (BOOL) start {
     _http.handleRedirects = NO;  // CFStream will handle redirects instead
 
     // Configure HTTP proxy -- CFNetwork makes us do this manually, unlike NSURLConnection :-p
-    NSDictionary* proxy = url.my_proxySettings;
-    if (proxy) {
-        LogTo(ChangeTracker, @"Changes feed using proxy %@", proxy);
-        bool ok = CFReadStreamSetProperty(cfInputStream, kCFStreamPropertyHTTPProxy,
-                                          (CFDictionaryRef)proxy);
+    CFDictionaryRef proxySettings = CFNetworkCopySystemProxySettings();
+    if (proxySettings) {
+        LogTo(ChangeTracker, @"Changes feed using proxy settings %@", proxySettings);
+        Boolean ok = CFReadStreamSetProperty(cfInputStream, kCFStreamPropertyHTTPProxy, proxySettings);
         Assert(ok);
     }
+    CFRelease(proxySettings);
```

Comment:
The above line should go inside the `if` block, to avoid potentially calling `CFRelease` on `NULL`.